src/crimson/common/config_proxy.cc

```diff
@@ -48,14 +62,36 @@ void ConfigProxy::show_config(ceph::Formatter* f) const {
 
 seastar::future<> ConfigProxy::parse_config_files(const std::string& conf_files)
 {
-  return do_change([this, conf_files](ConfigValues& values) {
-    const char* conf_file_paths =
-      conf_files.empty() ? nullptr : conf_files.c_str();
-      get_config().parse_config_files(values,
-                                      obs_mgr,
-                                      conf_file_paths,
-                                      &std::cerr,
-                                      CODE_ENVIRONMENT_DAEMON);
+  auto conffile_paths =
+    get_config().get_conffile_paths(*values,
+                                    conf_files.empty() ? nullptr : conf_files.c_str(),
+                                    &std::cerr,
+                                    CODE_ENVIRONMENT_DAEMON);
+  return seastar::do_with(std::move(conffile_paths), [this] (auto& paths) {
+    return seastar::repeat([path=paths.begin(), e=paths.end(), this]() mutable {
+      if (path == e) {
```

Comment:
`e` was a little bit confusing. How about `end` or `end_`?

Reply:
i'd avoid using the name collision. as `std::end()` is also named `end`. `e` is captured at line 71, and is used at line 72. after it's never used afterwards. so i think it is not worthy having a longer name.